common-files/utils/gas.mjs

```diff
@@ -0,0 +1,69 @@
+/* ignore unused exports */
+
+import axios from 'axios';
+import logger from './logger.mjs';
+
+/**
+ * Estimate the amount of gas that will be needed to submit a transaction (tx)
+ *
+ * The underlying RPC `eth_estimateGas` will check the balance of the sender,
+ * this means that even though the call doesn't consume any gas,
+ * the `from` address must have enough gas to execute the tx
+ *
+ * @async
+ * @function estimateGas
+ * @param {object} tx A tx object
+ * @param {Web3} web3
+ * @param {number} gasDefault Has default
+ * @param {number} gasMultiplier Buffer to apply to estimated gas - Has default
+ * @returns {Promise<number>}
+ */
+export async function estimateGas(tx, web3, gasDefault = 4000000, gasMultiplier = 2) {
+  logger.debug({ msg: 'Estimate gas for transaction...', tx });
+
+  let gas;
+  try {
+    gas = await web3.eth.estimateGas(tx);
+    logger.debug({ msg: 'Gas estimated at', gas });
+  } catch (error) {
+    gas = gasDefault;
+    logger.warn({ msg: 'Gas estimation failed, use default', gas });
+  }
+  return Math.ceil(gas * gasMultiplier);
+}
+
+/**
+ * Estimate gas price
+ *
+ * @async
+ * @function estimateGasPrice
+ * @param {string} gasEstimateEndpoint URL for querying gas prices
+ * @param {Web3} web3
+ * @param {number} gasPriceDefault Has default
+ * @param {number} gasPriceMultiplier Buffer to apply to estimated gas price - Has default
+ * @returns {Promise<number>}
+ */
+export async function estimateGasPrice(
```

Comment:
Same as above. Can you take a look at `web3.mjs` and substitute default gasPrice value by your function?

Reply:
#1286